library/ssl_tls.c

```diff
@@ -1984,6 +2009,25 @@ static int ssl_tls13_session_save( const mbedtls_ssl_session *session,
     memcpy( p, session->resumption_key, session->resumption_key_len );
     p += session->resumption_key_len;
 
+#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION) && \
+    defined(MBEDTLS_SSL_SESSION_TICKETS) && \
+    defined(MBEDTLS_SSL_CLI_C)
+    if( session->endpoint == MBEDTLS_SSL_IS_CLIENT )
+    {
+        MBEDTLS_PUT_UINT16_BE( hostname_len, p, 0 );
+        p += 2;
+        if ( hostname_len > 0 &&
+             session->hostname != NULL )
```

Comment:
```suggestion
        if ( hostname_len > 0 )
```

Reply:
Why remove the NULL check?

Reply:
Seems it's also about the case " whether there is empty servername only has "\0""

Reply:
hostname_len == 0 means NULL.
hostname_len == 1 means empty string.